keras_nlp/samplers/top_p_sampler.py

```diff
@@ -34,6 +34,8 @@ class TopPSampler(Sampler):
 
     Args:
         p: float, the `p` value of top-p.
+        k: int, defaults to None. If set, only sample from top `k` tokens.
```

Comment:
... If set, this argument defines a heuristic "top-k" cutoff applied before the "top-p" sampling. All logits not in the top `k` will be discarded, and the remaining set will be sorted to find a cutoff point for `p`. Setting this argument can significantly speed sampling by reducing the size of the sort.

Reply:
thanks!